File: Product/src/main/java/com/api/product/client/StoreClient.java
package com.api.product.client;

import java.time.Duration;
import java.util.UUID;

import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@RequiredArgsConstructor
public class StoreClient {

    private final WebClient storeWebClient;

    public boolean existsStore(UUID storeId) {
        try {
            Boolean exists = storeWebClient.get()
                    .uri("/stores/existStore/{id}", storeId)
                    .retrieve()
                    .bodyToMono(Boolean.class)
                    .timeout(Duration.ofSeconds(5))
                    .block();
            return Boolean.TRUE.equals(exists);
        } catch (Exception e) {
            return false;
        }
    }

    public String userRole(UUID userId, UUID storeId){
        try{
            String role = storeWebClient.get()
                            .uri("/stores/{storeId}/isOwner/{userId}", storeId, userId)
                            .retrieve()
                            .bodyToMono(String.class)
                            .timeout(Duration.ofSeconds(5))
                            .block();
            // Store devuelve un enum (StoreRole), que al serializarse a JSON queda como
            // un string entre comillas (ej. "OWNER"). bodyToMono(String.class) entrega
            // ese texto crudo tal cual, comillas incluidas, por lo que sin este strip
            // "OWNER".equals(role) nunca es true.
            if (role != null) {
                role = role.replace("\"", "");
            }
            log.info("[StoreClient.userRole] storeId={} userId={} role={}", storeId, userId, role);
            return role;
        } catch(Exception e){
            log.error("[StoreClient.userRole] storeId={} userId={} fallo: {}", storeId, userId, e.getMessage());
            throw new RuntimeException("Error: " + e.getMessage());
        }
    }
}

File: IaStoreBuilder/service/client/store_client.py
import httpx
from config.settings import STORE_SERVICE_URL


async def get_user_store_role(store_id: str, user_id: str) -> str | None:
    """GET /api/v1/stores/{storeId}/isOwner/{userId} → 'OWNER' | 'ADMIN' | 'STAFF' | None"""
    try:
        async with httpx.AsyncClient(timeout=5) as client:
            r = await client.get(
                f"{STORE_SERVICE_URL}/api/v1/stores/{store_id}/isOwner/{user_id}"
            )
            if r.status_code == 200:
                return r.json()
    except Exception as e:
        print(f"[store_client] role error: {e}")
    return None


async def get_store_info(store_id: str) -> dict:
    """GET /api/v1/stores/{storeId}"""
    try:
        async with httpx.AsyncClient(timeout=5) as client:
            r = await client.get(
                f"{STORE_SERVICE_URL}/api/v1/stores/{store_id}"
            )
            if r.status_code == 200:
                return r.json()
    except Exception as e:
        print(f"[store_client] info error: {e}")
    return {}


async def get_store_settings(store_id: str, jwt_token: str) -> dict:
    """GET /api/v1/stores/settings/getSettings"""
    try:
        async with httpx.AsyncClient(timeout=5) as client:
            r = await client.get(
                f"{STORE_SERVICE_URL}/api/v1/stores/settings/getSettings",
                headers={"X-Store-Id": store_id, "Authorization": f"Bearer {jwt_token}"}
            )
            if r.status_code == 200:
                return r.json()
    except Exception as e:
        print(f"[store_client] settings error: {e}")
    return {}
